Update Wallet.md (#573)
btcpayserver/btcpayserver-doc: docs/Wallet.md

# BTCPay Server Wallet

BTCPay Server has a built in, full-node reliant wallet that allows easier funds management. 

> Each [store](CreateStore.md) and cryptocurrency have a separate wallet displayed on the wallets page. A wallet can be accessed by clicking **Wallets** from the top menu. To access a particular wallet, click **Manage**.

Reviewer: ```suggestion
Each [store](CreateStore.md)'s configured cryptocurrency has a separate wallet displayed on the wallets page. A wallet can be accessed by clicking **Wallets** from the top menu. To access a particular wallet, click **Manage**.
```

### Transactions

An overview of the incoming (green), outgoing (red) and unconfirmed (grayed out) transactions displayed together with timestamps and balances, sorted by date. You can click on the tx ID to preview the transaction on the block explorer.

![Individual Wallet](./img/wallet/WallettTransactions.png)

### Send

The Send function allows spending of the funds from the BTCPay wallet.

![Send from the Wallet](./img/wallet/WalletSend.png)

#### Signing a transaction (spending)

To spend the funds, you are required to **sign** the transaction. Transactions can be signed with:

- HD Private key or mnemonic seed
- Wallet supporting PSBT
- Hardware Wallet
- Hot Wallet

##### Signing with HD Private Key or mnemonic seed

If you set up an [existing wallet with your BTCPay Server](WalletSetup/#use-an-existing-wallet), you can spend the funds by inputing your private key into an appropriate field. Make sure to set a proper `AccountKeyPath` in Wallet > Settings otherwise you won't be able to spend.

##### Signing with a wallet supporting PSBT

> PSBT (Partially Signed Bitcoin transactions) are supported and can be signed with PSBT compatible wallet.

Reviewer: ```suggestion
PSBT (Partially Signed Bitcoin transactions) are supported and can be signed with PSBT compatible wallets.
```

Check this tutorial on how to [sign a transaction with ColdCard Hardware Wallet](ColdCardWallet.md#spending-from-btcpay-server-wallet-with-coldcard-psbt) completely air-gaped.

> ##### Signing with a hardware wallets

Reviewer: ```suggestion
##### Signing with a hardware wallet
```

> BTCPay Server has a built in hardware wallet support allowing you to use your hardware wallet within BTCPay, without leaking information to third-party apps or servers.

Reviewer: ```suggestion
BTCPay Server has built-in hardware wallet support allowing you to use your hardware wallet with BTCPay, without leaking information to third-party apps or servers.
```

[Check instructions](Vault.md) on how to set up and sign with a [compatible hardware wallet](https://github.com/bitcoin-core/HWI#device-support).

##### Signing with a hot wallet

> If you [created a new wallet](CreateWallet.md) when setting up your store and enabled a [hot wallet](HotWallet.md) you can sign a transaction with a private key stored on your server.

Reviewer: ```suggestion
If you [created a new wallet](CreateWallet.md) when setting up your store and enabled it as a [hot wallet](HotWallet.md), you can sign a transaction with a private key stored on your server.
```

:::

#### Advanced Settings

Certain wallet features are available for advanced users. Toggle the `Advanced Settings` within the `Send` tab to preview them.

#### Don't create UTXO change

This option is available in the `Advanced mode` of the `Send` page.

It is a privacy enhancing feature which is useful when you're sending funds to another wallet of yours or to an exchange. It makes sure that no change UTXO is created by **rounding up** the amount sent.

By default this feature is disabled, so if your wallet has an UTXO of `1.1 BTC` and you input an amount equals to `1.0 BTC`, the resulting transaction will have two outputs `0.1 BTC` of change, and `1.0 BTC` to your destination.

Blockchain analysis will understand that those `0.1 BTC` of change belong to the same entity which controlled `1.1 BTC` before, and can track the future purchase you make under the same pattern.

By enabling this feature, BTCPay Server wallet will round up the amount sent to `1.1 BTC` such that no change output is sent back to you.

Warning: Despite the fact, in this example, that you entered `1.0` in the amount field, the amount that will really be sent to your destination will be `1.1 BTC`.

> ##### RBF (Replace-by-fee)

Author: @britttttk I know you wanted to document this, if you want you can commit it here.

Reviewer: There is some new settings around this in the next release. By default, it will now randomize it for more privacy.

Blockchain analysis will understand that those `0.1 BTC` of change belong to the same entity which controlled `1.1 BTC` before, and can track the future purchase you make under the same pattern.

By enabling this feature, BTCPay Server wallet will round up the amount sent to `1.1 BTC` such that no change output is sent back to you.

Warning: Despite the fact, in this example, that you entered `1.0` in the amount field, the amount that will really be sent to your destination will be `1.1 BTC`.

##### RBF (Replace-by-fee)

##### Coin Selection

Coin selection is a an advanced privacy-enhancing feature that allows you to specifically select coins that you would like to spend when crafting a transaction. For example, paying with coins that are fresh from a coinjoin mix.

To make the selection easier, coin-selection works natively with the wallet labels feature. This allows you to label any incoming funds for smoother UTXO management and spending.

##### Other features

###### Camera QR scan

> Scan option in wallet (camera icon) lets you use your device’s camera to scan an invoice when sending from the wallet. It auto-populates the invoice information so you don’t have to manually copy-paste address and amount.

Reviewer: ```suggestion
Scan option in wallet (camera icon in send screen) lets you use your device’s camera to scan a QR code containing an address or BIP21 payment link. It auto-populates the sending information so that you don’t have to manually copy-paste an address and amount.
```

**Wallet import**: When users add a derivation scheme of a wallet that had transactions in the past (previously used wallet), BTCPay won't be able to show the balance and transactions from the past.

![Wallet rescan progress](./img/wallet/WalletRescanProgress.png)

Re-scan is a feature that solves both of these problems. Once the scan is complete, BTCPay Server will show the correct balance, along with the past transactions of the wallet.

Wallet re-scan requires access to the full node which means that this function is only available for server owners.

Users who use a third party host should use a newly generated xpub key and also use an external wallet like Electrum which allows them to increase the gap limit.

### PSBT

> ### Settings

Author: @Kukks need your help with this one, especially the accountkeypath and fingerprint if you can document this section when you get time, would help a lot.

**Wallet import**: When users add a derivation scheme of a wallet that had transactions in the past (previously used wallet), BTCPay won't be able to show the balance and transactions from the past.

![Wallet rescan progress](./img/wallet/WalletRescanProgress.png)

Re-scan is a feature that solves both of these problems. Once the scan is complete, BTCPay Server will show the correct balance, along with the past transactions of the wallet.

Wallet re-scan requires access to the full node which means that this function is only available for server owners.

Users who use a third party host should use a newly generated xpub key and also use an external wallet like Electrum which allows them to increase the gap limit.

### PSBT 

> In the Partially Signed Bitcoin Transactions (PSBT) tab, you can upload and decode an earlier signed PSBT from external wallet and broadcast it via BTCPay Wallet. For more information on PSBT check this link.

Author: I know but for wallet only signed was useful for me, please commit if you want to extend it don't know what else to add tbh.
